packages/typespec-ts/src/modular/buildCodeModel.ts

```diff
@@ -1454,7 +1454,7 @@ function emitUnion(
     return {
       name: typeName,
       nullable: isNull,
-      description: sdkType.description || `Type of ${typeName}`,
+      description: sdkType.doc || `Type of ${typeName}`,
```

Comment:
Refer to : https://github.com/Azure/typespec-azure/pull/1560/files#diff-e186052723c0e3dfe654f3a7fd0f36bd8126c371ad017fdee5a8643ba3d8da10R234